src/ResourceManager/ContainerInstance/Commands.ContainerInstance/Commands/NewAzureContainerGroupCommand.cs

```diff
@@ -60,29 +56,25 @@ public class NewAzureContainerGroupCommand : ContainerInstanceCmdletBase
         public string Image { get; set; }
 
         [Parameter(
-            Mandatory = true,
-            ParameterSetName = CreateContainerGroupWithRegistryParamSet,
+            Mandatory = false,
             HelpMessage = "The custom container registry credential.")]
         [ValidateNotNullOrEmpty]
         public PSCredential RegistryCredential { get; set; }
 
         [Parameter(
-            Mandatory = true,
-            ParameterSetName = CreateContainerGroupWithAzureFileVolumeParamSet,
```

Comment:
Please change this into two parameter sets (it is better to do validation over parameter sets rather than throwing at runtime).  You should have: 
New-AzureRmContainerGroup
   [-ResourceGroupName] <String>
   [-Name] <String>
   [-Image] <String>
   [-RegistryCredential <PSCredential> ]
   [-Location <String>]
   [-OsType <String>]
   [-RestartPolicy <String>]
   [-Cpu <Int32>]
   [-MemoryInGB <Double>]
   [-IpAddressType <String>]
   [-Port <Int32[]>]
   [-Command <String>]
   [-EnvironmentVariable <Hashtable>]
   [-RegistryServerDomain <String>]
   [-Tag <Hashtable>]
   [-DefaultProfile <IAzureContextContainer>]
   [-WhatIf]
   [-Confirm]
   [<CommonParameters>]

and

New-AzureRmContainerGroup
   [-ResourceGroupName] <String>
   [-Name] <String>
   [-Image] <String>
   [ -RegistryCredential <PSCredential> ]
   -AzureFileVolumeShareName <String>
   -AzureFileVolumeAccountCredential <PSCredential>
   -AzureFileVolumeMountPath <String>
   [-Location <String>]
   [-OsType <String>]
   [-RestartPolicy <String>]
   [-Cpu <Int32>]
   [-MemoryInGB <Double>]
   [-IpAddressType <String>]
   [-Port <Int32[]>]
   [-Command <String>]
   [-EnvironmentVariable <Hashtable>]
   [-Tag <Hashtable>]
   [-DefaultProfile <IAzureContextContainer>]
   [-WhatIf]
   [-Confirm]
   [<CommonParameters>]

Reply:
A couple of notes, as I had not seen [this code](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/ContainerInstance/Commands.ContainerInstance/Models/ContainerGroupCreationParameters.cs#L162):

You should make Location mandatory instead of throwing an exception at runtime.
You should have these parameter sets to get rid of runtime validation over RegistryServer:
 New-AzureRmContainerGroup
[-ResourceGroupName] 
[-Name] 
[-Image] 
[-RegistryCredential ]
[-Location ]
[-OsType ]
[-RestartPolicy ]
[-Cpu ]
[-MemoryInGB ]
[-IpAddressType ]
[-Port <Int32[]>]
[-Command ]
[-EnvironmentVariable ]
[-Tag ]
[-DefaultProfile ]
[-WhatIf]
[-Confirm]

New-AzureRmContainerGroup
[-ResourceGroupName] 
[-Name] 
[-Image] 
-RegistryServerDomain
-RegistryCredential
[-Location ]
[-OsType ]
[-RestartPolicy ]
[-Cpu ]
[-MemoryInGB ]
[-IpAddressType ]
[-Port <Int32[]>]
[-Command ]
[-EnvironmentVariable ]
[-Tag ]
[-DefaultProfile ]
[-WhatIf]
[-Confirm]

The third validation you should keep, as it cannot be done through parameter set validation.  The fourth validation could be moved into parameter validation using validatescript, but this is not required.

Reply:
For location, it's not required. If not present, it will try to get the same location as the resource group.

Other than location, I'm confused. I think the problem is that I can't have 2 parameter sets here. In your 2nd comment, where is Azure File parameters then? When using Registry parameters, I should be still be able to use Azure File parameters.

Reply:
oh I see. So the Registry parameters should belong to 2 parameter sets and Azure File parameters should only belong to 1.

Reply:
Yes, that should eliminate all of the parameter validation that can be done on the parameter set level.

Reply:
I can't find any example using `ValidateScript` in c#. Do you have any example?

Reply:
The syntax should be pretty much the same for both PowerShell and C# syntax: https://powershell.org/2013/05/21/validatescript-for-beginners/.  Basically, you want the script to return true if the parameter is valid, and false otherwise.  Error Message can be given like this: https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.validatescriptattribute.errormessage?view=pscore-6.0.0&viewFallbackFrom=powershellsdk-1.1.0

Reply:
This is not a blocking change for your PR, so it is up to you if you want to make this change.

Reply:
Good to know. It seems it requires a recent version of powershell sdk.
Anyway, I add back parameter set for azure file.

Reply:
Can you also add the third parameter set we discussed above? Looks like you only have two at the moment.

Reply:
Ping on this comment